app/uk/gov/hmrc/exports/services/mapping/goodsshipment/WarehouseBuilder.scala

```diff
@@ -30,17 +30,20 @@ class WarehouseBuilder @Inject()() extends ModifyingBuilder[WarehouseIdentificat
     }
 
   private def isDefined(warehouse: WarehouseIdentification): Boolean =
-    warehouse.identificationNumber.isDefined && warehouse.identificationType.isDefined
+    warehouse.identificationNumber.isDefined
 
   private def createWarehouse(data: WarehouseIdentification): Warehouse = {
     val warehouse = new Warehouse()
 
+    val warehouseIdentificationType = data.identificationNumber.map(_.substring(0,1)).getOrElse("")
```

Comment:
Hmm, I'm just wondering if taking `headOption` and `tail` will not be enough.

You're just extracting one character (first one)